src/server/game/AI/CoreAI/ConversationAI.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_CONVERSATIONAI_H
+#define TRINITY_CONVERSATIONAI_H
+
+#include "Define.h"
+#include "ObjectGuid.h"
+
+class Conversation;
+class Unit;
+class Player;
+
+class TC_GAME_API ConversationAI
+{
+        uint32 _scriptId;
```

Comment:
_scriptId is a private member, but convo is protected, do we not want to make convo private as well and ensure people use getters to access it?

uint32 GetScriptId() const { return _scriptId; } (rather than GetId only)
Conversation* GetConversation() const { return convo; }

Reply:
thats intended to enable `convo` usage similar to `me` or `at` in CreatureAI/AreaTriggerAI

Reply:
oh, I c